proposals/42-human-readable-mutable-names.md

```diff
@@ -0,0 +1,235 @@
+-->
+
+We see CID, IPNS record, and DNSLink being de-facto standard for cryptographically-verifiable content-addressing and human-readable names on the distributed web. 
+
+There are three distinct types of addresses:
+  - immutable content:  `/ipfs/{cid}`
+  - mutable content
+      - cryptographic, IPNS record signed with a specific libp2p key: `/ipns/{libp2p-key-cid}` 
+      - human-readable, [DNSLink](https://dnslink.io): `/ipns/{dns-name}`
+        (which in turn, resolves to either an immutable or mutable content path from DNS TXT record)
+
+
+**There are known gaps in the mutable story:**
+- IPNS records are only published by the original author, which severely limits the utility of our mutable names
+    - It is not possible to pin/republish `/ipns/{libp2p-key-cid}` names by other peers, so when original publisher disappears, IPNS links no longer resolve and we see "IPNS rot". 
+    - **Due to IPNS rot people simply don't use it:**
+        - Instead publishing updates via `ipfs name publish {cid}`, website operators choose to constantly update DNSLink's DNS TXT record
+            - They point at reliable immutable paths (`dnslink=/ipfs/{cid}`) (no IPNS record means it resolves fast and never break).
+            - It works around the need for running a service responsible for constant republishing of IPNS record.
+            - This is a pragmatic choice, however:
+              - adds admin related to DNS updates (need to learn DNS provider's API for automating this, which introduces soft vendor-lock-in)
+              - most likely not the best choice for security-sensitive use cases where human-readable names are not required: means DNS is used in contexts where cryptographic addressing should be used instead.
+- **OS-level DNS resolver is used for all DNS TXT record lookups.** 
+  - IPFS node is unable to control which DNS resolver is used for specific TLD, all queries go to global DNS server configured in the operating system
+    - This harms competition and slows down innovation in the decentralized naming space. For example, if user changes their DNS resolver to one that is capable of resolving all TLDs over ENS, they are unable to resolve TLDs from OpenNIC or UnstoppableDomains. 
+  - Given how DNS is implemented in OS and ISPs, queries are most likely sent in plain text and can be spoofed in LAN or MITMd at the ISP level. At the time of writing this proposal DNS over HTTPS is not supported, and it is not trivial for regular user to set up a custom DNS server to change this behavior.
+
+**TLDR:** **We need to fix and future-proof the way mutable names work in our stack:**
+- Make IPNS useful on its own, and inside of DNSLink records.
+    - DNS TXT record should be set only once, updates should be published over IPNS.
```

Comment:
This is https://github.com/protocol/web3-dev-team/pull/19 right?

Reply:
Yes, this is overlap with fixing IPNS.